backend/routes/user.js

```diff
@@ -0,0 +1,124 @@
+const express = require('express');
+const router = express.Router();
+const gravatar = require('gravatar');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const passport = require('passport');
+const validateRegisterInput = require('../validation/register');
+const validateLoginInput = require('../validation/login');
+
+const User = require('../models/User');
+
+router.post('/register', function(req, res) {
+
+  const {
+    errors,
+    isValid
+  } = validateRegisterInput(req.body);
+
+  if (!isValid) {
+    console.log("error in 20 lines of code user.js")
+    return res.status(400).json(errors);
+  }
+  User.findOne({
+    email: req.body.email
+  }).then(user => {
+    if (user) {
+      return res.status(400).json({
+        email: 'Email already exists'
+      });
+    } else {
+      const avatar = gravatar.url(req.body.email, {
+        s: '200',
+        r: 'pg',
+        d: 'mm'
+      });
+      const newUser = new User({
+        name: req.body.name,
+        email: req.body.email,
+        password: req.body.password,
+        avatar
+      });
+
+      bcrypt.genSalt(10, (err, salt) => {
+        if (err) console.error('There was an error', err);
+        else {
+          bcrypt.hash(newUser.password, salt, (err, hash) => {
+            if (err) console.error('There was an error', err);
+            else {
+              newUser.password = hash;
+              newUser
+                .save()
+                .then(user => {
+                  res.json(user)
+                });
+            }
+          });
+        }
+      });
+    }
+  });
+});
+
+router.post('/login', (req, res) => {
+
+  const {
+    errors,
+    isValid
+  } = validateLoginInput(req.body);
+
+  if (!isValid) {
+    console.log("71 lines of code")
+    return res.status(400).json(errors);
+  }
+
+  const email = req.body.email;
+  const password = req.body.password;
+
+  User.findOne({
```

Comment:
you already used this method in previous callback

pls move to separate file and method if it is reusable.
For example, dbQuery folder, user file, and method findByEmail or something like this